airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py

```diff
@@ -110,21 +110,20 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
             self.validate_replication_dates(config)
             self.validate_stream_report_options(config)
             stream_kwargs = self._get_stream_kwargs(config)
-            orders_stream = Orders(**stream_kwargs)
-            next(orders_stream.read_records(sync_mode=SyncMode.full_refresh))
+
+            if config.get("account_type", "Seller") == "Seller":
```

Comment:
According to this, all existing users forced to Seller account?

Reply:
No, the check command will check access to the Vendor report stream for the Vendor account type. The default value was added to avoid errors for the config without account type, although account type is the required property with the default value of `Seller`.

Reply:
So, if all users who set up connections before these changes will be forced to be Sellers because they do not have an `account_type` specified in their credentials.

Reply:
No, actually, all the users should already have `account_type` in their config as it is [used for the Oauth](https://github.com/airbytehq/airbyte/blob/a6f4c2e11dd820e8a7d5532eedefa92c57d96c3a/airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/spec.json#L218), also [this parameter is set during the migration from the previous version](https://github.com/airbytehq/airbyte/blob/7db133b66c7a24830cf4b04c5a329e84ec754e8a/airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/config_migrations.py#L30).